src/wait-for.js

```diff
@@ -125,6 +141,10 @@ function waitFor(
     }
 
     function checkRealTimersCallback() {
+      clearTimeout(intervalId)
```

Comment:
`clearTimeout` on something called "interval" seems off. Either the method is wrong or the naming is off.